app/assets/javascripts/directives/containers/chart-data-mixin.js

```diff
@@ -0,0 +1,102 @@
+angular.module('containers').factory('ChartsDataMixin', ['$timeout', '$q', function chartDataMixinFactory ($timeout, $q) {
+
+    var sparklineTimeTooltip = function(d) {
+        return '  <table class="c3-tooltip">' +
+            '    <tbody>' +
+            '      <td class="value">' +  d[0].x.toLocaleTimeString() + '</td>' +
+            '      <td class="value text-nowrap">' +  d[0].value + ' ' + d[0].name + '</td>' +
+            '    </tbody>' +
+            '  </table>';
```

Comment:
I'd much rather see a directive handling that, does it have to be like this? (It seems tooltipFn is not actually used anywhere, so I have no idea what the intended use is...)

But at the very least, we could do something like

```js
var col1 = $('<td>').addClass('value').text( d[0].x.toLocaleTimeString() );
var col2 = $('<td>').addClass('value').addClass('text-nowrap').text( d[0].value + ' ' + d[0].name );
return $('<table>').addClass('c3-tooltip').append(col1, col2).html();
```

I guess it's less pretty but definitely safe. Or whatever babel outputs when you use an ES6 template string.

Oh! Or actually, we do have lodash and [_.template](https://lodash.com/docs#template), so you *could* do something like

```js
return _.template('<table class="c3-tooltip">' +
            '    <tbody>' +
            '      <td class="value"><%- col1 %></td>' +
            '      <td class="value text-nowrap"><%- col2 %></td>' +
            '    </tbody>' +
            '  </table>')({ col1: d[0].x.toLocaleTimeString(), col2: d[0].value + ' ' + d[0].name });
```